core/frontend/src/ViewState.ts

```diff
@@ -364,15 +374,15 @@ export abstract class ViewState extends ElementState {
   public abstract get details(): ViewDetails;
 
   /** Returns true if this ViewState is-a [[ViewState3d]] */
-  public is3d(): this is ViewState3d { return this instanceof ViewState3d; }

```

Comment:
should these be abstract?

Reply:
or:
```ts
public abstract is3d(): boolean;
public is2d(): boolean { return !this.is3d(); }
```

Reply:
Probably.

Reply:
Pushed that change.

Reply:
Please resolve comments if changes/responses satisfactory.